dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java

```diff
@@ -230,7 +230,24 @@ public static int getMinutesUntilPasswordResetTokenExpires() {
      * by the Settings Service configuration.
      */
     public String getDataverseSiteUrl() {
-        return getDataverseSiteUrlStatic();
+        String hostUrl = System.getProperty(SITE_URL);
+        if (hostUrl != null && !"".equals(hostUrl)) {
+            return hostUrl;
+        }
+        String hostName = SystemConfig.FQDN;
+        if (hostName == null) {
+            try {
+                hostName = InetAddress.getLocalHost().getCanonicalHostName();
+            } catch (UnknownHostException e) {
+                return null;
+            }
+        }
+        hostUrl = "https://" + hostName;
+        return hostUrl;
+    }
+
+    public String getFqdnProperty() {
+        return System.getProperty(SystemConfig.FQDN);
```

Comment:
The FQDN should be handled the same way as in getDataverseSiteUrl().

Reply:
What do you mean? We only want to get the property, for now by getProperty() so we can switch to properties file in the future. We don't want to 'handle' it in any way unless we change system logic.